src/uu/yes/src/yes.rs

```diff
@@ -71,11 +73,11 @@ fn prepare_buffer<'a>(input: &'a str, buffer: &'a mut [u8; BUF_SIZE]) -> &'a [u8
 
 #[cfg(unix)]
 fn enable_pipe_errors() -> Result<()> {
-    let ret = unsafe { libc::signal(libc::SIGPIPE, libc::SIG_DFL) };
-    if ret == libc::SIG_ERR {
-        return Err(io::Error::new(io::ErrorKind::Other, ""));
+    // SAFETY: this function is safe as long as we do not use a custom SigHandler -- we use the default one.
+    match unsafe { signal(SIGPIPE, SigDfl) } {
```

Comment:
Nix `From<Errno>` in implemented by `io::Result` so you could use that for the conversion too.

Reply:
Another option is returning `UResult` because both `io::Result` and `nix::Errno` can be converted into that.